Core/Battler.gd

```diff
@@ -148,16 +148,18 @@ func _handle_enemy_deaths() -> void:
 	for enemy: Enemy in _enemy_list:
 		enemy.get_party_component().set_party(_enemy_list)
 
+
 ## return TRUE if battle have ended either with victory or defeat
 func _check_and_handle_battle_end() -> bool:
-	if PlayerManager.player.get_health_component().current_health == 0:
-		PhaseManager.on_combat_end.emit(GlobalEnums.CombatResult.DEFEAT)
+	if PlayerManager.player.get_health_component().current_health == 0: # ? change this to <= 0
```

Comment:
Yes the value can't go below 0 because of the line 43 in Entity/Component/HealthComponent:
```GDScript
var new_health: float = clampf(current_health - damage, 0, max_health)
```
But we never know, just to be sure isn't a bad approach